File: Stack/Stack-Python/src/Stack.py
from Node import Node
from Data import Data

class Stack:
    def __init__(self):
        self.top=None
        self.size=0

    def push(self,word):
        newNode=Node(Data(word))
        newNode.next=self.top
        self.top=newNode
        self.size+=1

    def peek(self):
        if self.isEmpty():
            return
        print(self.top.data.word)

    def pop(self):
        if self.isEmpty():
            return
        self.top=self.top.next
        self.size-=1

    def isEmpty(self):
        return self.top is None

File: Stack/Stack-Csharp/src/Stack.cs
using System;

namespace src 
{
  class Stack 
  {
    public Stack(){this.top=null;}

    public void push(string data)
    {
      Node newNode=new Node(new Data(data));
      newNode.next=this.top;
      this.top=newNode;
    }

    public void pop()
    {
      if(isEmpty())
      {
        Console.WriteLine("The stack is empty");
        return;
      }

      this.top=this.top.next;
    }

    public void peek()
    {
      if(this.isEmpty())
      {
        Console.WriteLine("The stack is empty"); 
        return;
      }

      Console.WriteLine(this.top.data.data);
    }

    private bool isEmpty()
    {
      return top==null;
    }

    private Node top;
  }
}
